### DISCUSSION TITLE
Error when running monailabel with bundle model nodule detection

### DISCUSSION THREAD

**Discussion by monaalsh** [Project-MONAI/MONAILabel#1429] [General]
I get the following error after running monailabel with bundle model nodule detection

`monailabel start_server --app apps/monaibundle --studies datasets/Task06_Lung/imagesTr --conf models lung_nodule_ct_detection`

below in the error I get

> RuntimeError: Failed to instantiate {'_target_': 'RetinaNet', 'spatial_dims': 3, 'num_classes': 1, 'num_anchors': 3, 'feature_extractor': BackboneWithFPN(

> [2023-05-10 08:13:15,982] [70122] [MainThread] [ERROR] (uvicorn.error:56) - Application startup failed. Exiting.

**Comment by diazandr3s** [comment]
Hi @monaalsh,

Thanks for reporting this.
Would you please attach the full logs? Which MONAI Label version are you using?

Thanks!

**Comment by monaalsh** [comment]
hi @diazandr3s 

Here is the log 

The version of monailabel is 
```

MONAILabel version: 0.6.0
Numpy version: 1.24.3
Pytorch version: 1.13.1+cu117
MONAILabel rev id: 40fd783a704f4d6a0baa8d34086dd336efdf1e6f
```

The log when running the nodule detection model is below, it doesn't happen when running other model such as the default one in the 
quickstart page for Spleen segmentation.
Thank you in advance.



```
TypeError: __init__() got an unexpected keyword argument 'use_list_output'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/mona/.local/lib/python3.8/site-packages/monai/bundle/config_item.py", line 283, in instantiate
    return instantiate(modname, **args)
  File "/home/mona/.local/lib/python3.8/site-packages/monai/utils/module.py", line 245, in instantiate
    raise RuntimeError(f"Failed to instantiate '{path}' with kwargs: {kwargs}") from e
RuntimeError: Failed to instantiate 'monai.apps.detection.networks.retinanet_network.RetinaNet' with kwargs: {'spatial_dims': 3, 'num_classes': 1, 'num_anchors': 3, 'feature_extractor': BackboneWithFPN(
  (body): IntermediateLayerGetter(
    (conv1): Conv3d(1, 64, kernel_size=(7, 7, 7), stride=(2, 2, 1), padding=(3, 3, 3), bias=False)
    (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (relu): ReLU(inplace=True)
    (maxpool): MaxPool3d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
    (layer1): Sequential(
      (0): ResNetBottleneck(
        (conv1): Conv3d(64, 64, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (downsample): Sequential(
          (0): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1))
          (1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): ResNetBottleneck(
        (conv1): Conv3d(256, 64, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
      (2): ResNetBottleneck(
        (conv1): Conv3d(256, 64, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
    )
    (layer2): Sequential(
      (0): ResNetBottleneck(
        (conv1): Conv3d(256, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (downsample): Sequential(
          (0): Conv3d(256, 512, kernel_size=(1, 1, 1), stride=(2, 2, 2))
          (1): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): ResNetBottleneck(
        (conv1): Conv3d(512, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
      (2): ResNetBottleneck(
        (conv1): Conv3d(512, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
      (3): ResNetBottleneck(
        (conv1): Conv3d(512, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
    )
  )
  (fpn): FeaturePyramidNetwork(
    (inner_blocks): ModuleList(
      (0): Conv3d(256, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1))
      (1): Conv3d(512, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1))
    )
    (layer_blocks): ModuleList(
      (0): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1))
      (1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1))
    )
    (extra_blocks): LastLevelMaxPool(
      (maxpool): MaxPool3d(kernel_size=1, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
  )
), 'size_divisible': [16, 16, 8], 'use_list_output': False}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/mona/.local/lib/python3.8/site-packages/starlette/routing.py", line 635, in lifespan
    async with self.lifespan_context(app):
  File "/home/mona/.local/lib/python3.8/site-packages/starlette/routing.py", line 530, in __aenter__
    await self._router.startup()
  File "/home/mona/.local/lib/python3.8/site-packages/starlette/routing.py", line 612, in startup
    await handler()
  File "/home/mona/.local/lib/python3.8/site-packages/monailabel/app.py", line 106, in startup_event
    instance = app_instance()
  File "/home/mona/.local/lib/python3.8/site-packages/monailabel/interfaces/utils/app.py", line 51, in app_instance
    app = c(app_dir=app_dir, studies=studies, conf=conf)
  File "/home/mona/Documents/Lung_cancer/apps/monaibundle/main.py", line 47, in __init__
    super().__init__(
  File "/home/mona/.local/lib/python3.8/site-packages/monailabel/interfaces/app.py", line 101, in __init__
    self._infers = self.init_infers()
  File "/home/mona/Documents/Lung_cancer/apps/monaibundle/main.py", line 62, in init_infers
    i = BundleInferTask(b, self.conf)
  File "/home/mona/.local/lib/python3.8/site-packages/monailabel/tasks/infer/bundle.py", line 108, in __init__
    network = self.bundle_config.get_parsed_content(self.const.key_network_def(), instantiate=True)
  File "/home/mona/.local/lib/python3.8/site-packages/monai/bundle/config_parser.py", line 271, in get_parsed_content
    return self.ref_resolver.get_resolved_content(id=id, **kwargs)
  File "/home/mona/.local/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 185, in get_resolved_content
    return self._resolve_one_item(id=id, **kwargs)
  File "/home/mona/.local/lib/python3.8/site-packages/monai/bundle/reference_resolver.py", line 163, in _resolve_one_item
    self.resolved_content[id] = item.instantiate() if kwargs.get("instantiate", True) else item
  File "/home/mona/.local/lib/python3.8/site-packages/monai/bundle/config_item.py", line 285, in instantiate
    raise RuntimeError(f"Failed to instantiate {self}.") from e
RuntimeError: Failed to instantiate {'_target_': 'RetinaNet', 'spatial_dims': 3, 'num_classes': 1, 'num_anchors': 3, 'feature_extractor': BackboneWithFPN(
  (body): IntermediateLayerGetter(
    (conv1): Conv3d(1, 64, kernel_size=(7, 7, 7), stride=(2, 2, 1), padding=(3, 3, 3), bias=False)
    (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (relu): ReLU(inplace=True)
    (maxpool): MaxPool3d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
    (layer1): Sequential(
      (0): ResNetBottleneck(
        (conv1): Conv3d(64, 64, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (downsample): Sequential(
          (0): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1))
          (1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): ResNetBottleneck(
        (conv1): Conv3d(256, 64, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
      (2): ResNetBottleneck(
        (conv1): Conv3d(256, 64, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(64, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
    )
    (layer2): Sequential(
      (0): ResNetBottleneck(
        (conv1): Conv3d(256, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (downsample): Sequential(
          (0): Conv3d(256, 512, kernel_size=(1, 1, 1), stride=(2, 2, 2))
          (1): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): ResNetBottleneck(
        (conv1): Conv3d(512, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
      (2): ResNetBottleneck(
        (conv1): Conv3d(512, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
      (3): ResNetBottleneck(
        (conv1): Conv3d(512, 128, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (conv3): Conv3d(128, 512, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
        (bn3): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
      )
    )
  )
  (fpn): FeaturePyramidNetwork(
    (inner_blocks): ModuleList(
      (0): Conv3d(256, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1))
      (1): Conv3d(512, 256, kernel_size=(1, 1, 1), stride=(1, 1, 1))
    )
    (layer_blocks): ModuleList(
      (0): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1))
      (1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1))
    )
    (extra_blocks): LastLevelMaxPool(
      (maxpool): MaxPool3d(kernel_size=1, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
  )
), 'size_divisible': [16, 16, 8], 'use_list_output': False}.

[2023-05-10 15:27:55,304] [353705] [MainThread] [ERROR] (uvicorn.error:56) - Application startup failed. Exiting.
```

**Comment by junxiant** [comment]
Hello,

I faced the same issue earlier, you have to update your monailabel to the rc version

```
pip install monailabel==0.7.0rc6
```

worked for me

**Comment by diazandr3s** [comment]
Thanks, @junxiant.

@monaalsh could you please update the MONAI Label and let us know if that works for you?

Thanks, both!

**Comment by diazandr3s** [comment]
https://github.com/Project-MONAI/MONAILabel/issues/1430#issuecomment-1544060367

**Comment by wrecheis** [comment]
Since I am pretty new in monailabel I may have missed something essential in this case. I testet pancreas dint CT detection, that worked perfectly smooth (with own data) with 3D Slicer. But the bundle nodule detection throws the following error (output of monai server debug mode):
2024-01-09 09:29:42,911] [117108] [MainThread] [ERROR] (uvicorn.error:434) - Exception in ASGI application
Traceback (most recent call last):
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/utils/module.py", line 261, in instantiate
    return component(**kwargs)
TypeError: __init__() got an unexpected keyword argument 'meta_key_postfix'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/bundle/config_item.py", line 293, in instantiate
    return instantiate(modname, mode, **args)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/utils/module.py", line 271, in instantiate
    raise RuntimeError(
RuntimeError: Failed to instantiate component 'monai.transforms.utility.dictionary.EnsureChannelFirstd' with kwargs: {'keys': 'image', 'meta_key_postfix': 'meta_dict'}
 set '_mode_=debug' to enter the debugging mode.
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/fastapi/routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monailabel/endpoints/infer.py", line 206, in api_run_inference
    return run_inference(background_tasks, model, image, session_id, params, file, label, output)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monailabel/endpoints/infer.py", line 172, in run_inference
    result = instance.infer(request)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monailabel/interfaces/app.py", line 307, in infer
    result_file_name, result_json = task(request)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monailabel/tasks/infer/basic_infer.py", line 297, in __call__
    pre_transforms = self.pre_transforms(data)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monailabel/tasks/infer/bundle.py", line 201, in pre_transforms
    c = self.bundle_config.get_parsed_content(k, instantiate=True)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/bundle/config_parser.py", line 290, in get_parsed_content
    return self.ref_resolver.get_resolved_content(id=id, **kwargs)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/bundle/reference_resolver.py", line 193, in get_resolved_content
    return self._resolve_one_item(id=id, **kwargs)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/bundle/reference_resolver.py", line 163, in _resolve_one_item
    self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
  File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/bundle/reference_resolver.py", line 171, in _resolve_one_item
    self.resolved_content[id] = item.instantiate() if kwargs.get("instantiate", True) else item


File "/home/wolfgang/anaconda3/envs/monai/lib/python3.9/site-packages/monai/bundle/config_item.py", line 295, in instantiate
    raise RuntimeError(f"Failed to instantiate {self}") from e
RuntimeError: Failed to instantiate ConfigComponent: 
{'_target_': 'EnsureChannelFirstd',
 'keys': 'image',
 'meta_key_postfix': 'meta_dict'}

My setup is:
monai 1.3.0
monailabel 0.8.1

Many thanks for any suggestion!

**Reply by diazandr3s** [reply]
Hi @wrecheis,

Can you please share the command you used to start the server? What bundle specifically are you referring to?

**Comment by wrecheis** [comment]
Hi @diazandr3s , 
my command I used is the following:
monailabel start_server --app ./monaibundle --studies /home/wolfgang/daten/ct-lunge/lunge-CA/  --conf models lung_nodule_ct_detection

Changing --conf models to pancreas_ct_dints_segmentation works perfectly smooth
Again many thanks

**Comment by zhanghaorui** [comment]
@monaalsh Dear, I'm having the same problem as you. Have you solved it? 


monailabel -v
Using PYTHONPATH=/Users/zhanghaorui:

MONAILabel version: 0.8.1
Numpy version: 1.25.2
Pytorch version: 2.0.1
MONAILabel rev id: 5ef153f644796bc3983da34fce5f7863b4767d36

**Comment by wrecheis** [comment]
Hi, sorry for my late answer: In my case I had to delete the line "meta_key_postfix": "meta_dict" in the inference.json file (approx line number 100 in my editor) of the lung_nodule_detection module. After this everything runs smooth. I really wonder why this error happens or were there any changes after the release in monai model zoo? Maybe an update is necessary? Thanks anyway for the great work!

**Reply by diazandr3s** [reply]
Hi @wrecheis,

Thanks for the update. It's good to hear it is working on your end. Sorry, I'm just catching up on these issues.
Do you remember exactly which line was that? I can't see that on the latest bundle version: https://github.com/Project-MONAI/model-zoo/blob/dev/models/lung_nodule_ct_detection/configs/inference.json

Let us know,

**Comment by wrecheis** [comment]
Hi, I downloaded the stuff from https://monai.io/model-zoo. I did not check for updated versions on github - my fault. On you github version the line is already deleted. Anyway many thanks
